repo/blob/gcs/gcs_internal_test.go

```diff
@@ -0,0 +1,69 @@
+// Package gcs implements Storage based on Google Cloud Storage bucket.
+package gcs
+
+import (
+	"context"
+	"io/fs"
+	"os"
+	"testing"
+
+	gcsclient "cloud.google.com/go/storage"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGCSStorageCredentialsHelpers(t *testing.T) {
+	ctx := context.Background()
+	scope := gcsclient.ScopeReadOnly
+	var fileMode fs.FileMode = 0644
+
+	// test tokenSourceFromCredentialsJSON with service account key
+	gsaKeyServiceAccount := `{
+		"type": "service_account",
+		"project_id": "kopia-test-project",
+		"private_key_id": "kopia-test",
+		"private_key": "some-private-key",
+		"client_email": "kopia-test@developer.gserviceaccount.com",
+		"client_id": "kopia-test.apps.googleusercontent.com",
+		"auth_uri": "https://accounts.google.com/o/oauth2/auth",
+		"token_uri": "http://localhost:8080/token"
+	  }`
+
+	ts, err := tokenSourceFromCredentialsJSON(ctx, []byte(gsaKeyServiceAccount), scope)
+	require.NoError(t, err)
+	require.NotNil(t, ts)
+	// test tokenSourceFromCredentialsFile with service account key
+	gsaKeyServiceAccountFileName := "service-acount.json"
+	err = os.WriteFile(gsaKeyServiceAccountFileName, []byte(gsaKeyServiceAccount), fileMode)
+	require.NoError(t, err)
+	defer os.Remove(gsaKeyServiceAccountFileName)
```

Comment:
use `t.Cleanup(func () {....})` instead of `defer` here.

Reply:
Same here.

Reply:
Pushed changes